### PR TITLE
Message History - more

### PR DESCRIPTION
**Pull request by tuddman**
## Summary

- more concise wording for some functions and variables
- makes writing `group`s and `messages` to a file generic over `serde::Serialize`
- encapsulates generating the history reply into one callable function.
- adds a few more tests

### CODE REVIEW

File: xmtp_mls/src/groups/message_history.rs [hunk 1]
```diff
@@ -797,6 +884,22 @@ mod tests {
...
+    async fn test_prepare_history_reply() {
+        let wallet = generate_local_wallet();
+        let amal_a = ClientBuilder::new_test_client(&wallet).await;
+        let amal_b = ClientBuilder::new_test_client(&wallet).await;
+        assert_ok!(amal_b.allow_history_sync().await);
+
+        amal_a.sync_welcomes().await.expect("sync_welcomes");
+
+        let request_id = new_request_id();
+        let url = "http://0.0.0.0:5558/upload";
```
Comment (tuddman) [inline xmtp_mls/src/groups/message_history.rs:900]: ignore this for now.  I was testing with a locally running `message-history-server`; I'll fix this test in a minute..

Comment (tuddman) [reply xmtp_mls/src/groups/message_history.rs]: the upload server is now mocked in this test.

File: xmtp_mls/src/groups/message_history.rs [hunk 2]
```diff
@@ -182,7 +184,45 @@ where
...
 
-    pub(crate) async fn prepare_groups_to_sync(&self) -> Result<Vec<StoredGroup>, StorageError> {
+    async fn prepare_history_reply(
+        &self,
+        request_id: &str,
+        url: &str,
+    ) -> Result<HistoryReply, MessageHistoryError> {
+        let (history_file, enc_key) = self.write_history_bundle().await?;
+
+        let signing_key = HistoryKeyType::new_chacha20_poly1305_key();
```
Comment (neekolas) [inline xmtp_mls/src/groups/message_history.rs:194]: This is an encryption key, not a signing key, right?

Comment (neekolas) [inline xmtp_mls/src/groups/message_history.rs:194]: We should use a different function name for this if it's not actually returning a chacha20 key used for encryption

Comment (tuddman) [reply xmtp_mls/src/groups/message_history.rs:194]: fixed to actually use the `chacha20` cipher in [9896f89](https://github.com/xmtp/libxmtp/pull/791/commits/9896f892ae2d30776d9b170635c5bbafe32c12cb)

Comment (neekolas) [reply xmtp_mls/src/groups/message_history.rs:194]: It doesn't really matter since both are just a bunch of random bytes, but what I was getting at is that you aren't using it as a chacha20 key. You're using it as a MAC key.

**Review by neekolas — APPROVED** [review]
